dist/dashcore-lib.js

```diff
@@ -55940,7 +55938,7 @@ ProRegTxPayload.prototype.toBuffer = function toBuffer(options) {
     .write(Buffer.from(this.scriptPayout, 'hex'))
     .write(Buffer.from(this.inputsHash, 'hex'));
 
-  if (!skipSignature) {
+  if (!skipSignature && this.payloadSig) {
```

Comment:
Just a question: is `skipSignature` useful? What's the case for having a signature but wanting to ignore it?

Reply:
I think yes it is indeed necessary, as you will need to skip the signature in case you want to create a proRegTx without an already existing collateral (cf. `protx register_fund` in dash-cli). In that case you spend 1000 Dash (include and sign inputs which combined equal 1000) to a new collateral address and skip the extra payload signature.

Reply:
Actually, I'm not so sure anymore. As you say you either set a signature or not. The only case where this would help would be if somebody would set a signature and then invalidate its existence by also setting `skipSignature`. But he could do the same by simply not setting the signature :).

Reply:
That can be in a further PR if we think it's not as helpful anymore. This one's good to go.

Reply:
@Cofresi @KamuelaFranco 
Skipping signature in serialization is needed if you're trying to re-sign a payload - otherwise, you'll sign a buffer that already has another signature in it, which will result in an invalid signature

Reply:
@Cofresi @KamuelaFranco 
Hmm, not sure if my explanation is good enough. Here is how the signing process is happening:
1. You serialize payload to a binary representation _that does not include_ the signature;
2. You sign the data you've got on step 1.

In case if you've modified the payload after you've signed it, you need to re-sign it. If you serialize the payload to a binary, you'll get the binary that already includes the signature. If you try to sign binary with already included signature, you'll get a signature that signs payload _and_ the old signature, but since the old signature isn't in the payload anymore, this will make the new signature unverifiable. Verification process verifies a signature only against an unsigned binary, which you can get if you skip signature when serializing.

Reply:
Great explanation, ghank you @antouhou!